clidocstool_md.go

```diff
@@ -179,10 +179,14 @@ func mdCmdOutput(cmd *cobra.Command, old string) (string, error) {
 			}
 			name += "`"
 			name = mdMakeLink(name, f.Name, f, isLink)
-			fmt.Fprintf(b, "%s | %s |\n", name, f.Usage)
+			fmt.Fprintf(b, "%s | %s |\n", mdEscapePipe(name), mdEscapePipe(f.Usage))
 		})
 		fmt.Fprintln(b, "")
 	}
 
 	return b.String(), nil
 }
+
+func mdEscapePipe(s string) string {
+	return strings.ReplaceAll(s, `|`, `\|`)
```

Comment:
Do you think we need to check if it's not already escaped? (thinking out loud if we can have that situation; perhaps it's not a problem)

Reply:
@thaJeztah Yeah I thought about that and if someone already escapes in the usage field he would look bad in the CLI so I don't think it's necessary to check that here:

```
docker buildx build --help

Usage:  docker buildx build [OPTIONS] PATH | URL | -

Start a build

Aliases:
  build, b

Options:
      --add-host strings                              Add a custom host-to-IP mapping (host:ip)
      --allow strings                                 Allow extra privileged entitlement, e.g. network.host, security.insecure
      --build-arg stringArray                         Set build-time variables
      --builder string                                Override the configured builder instance
      --cache-from stringArray                        External cache sources (eg. user/app:cache, type=local,src=path/to/dir)
      --cache-to stringArray                          Cache export destinations (eg. user/app:cache, type=local,dest=path/to/dir)
  -f, --file string                                   Name of the Dockerfile (Default is 'PATH/Dockerfile')
      --iidfile string                                Write the image ID to the file
      --label stringArray                             Set metadata for an image
      --load                                          Shorthand for --output=type=docker
      --metadata-file string                          Write build result metadata to the file
      --network string                                Set the networking mode for the RUN instructions during build (default "default")
      --no-cache                                      Do not use cache when building the image
  -o, --output stringArray                            Output destination (format: type=local,dest=path)
      --platform stringArray                          Set target platform for build
      --progress string                               Set type of progress output (auto, plain, tty). Use plain to show container output (default "auto")
      --pull                                          Always attempt to pull a newer version of the image
      --push                                          Shorthand for --output=type=registry
      --secret stringArray                            Secret file to expose to the build: id=mysecret,src=/local/secret
      --ssh default\|<id>[=<socket>\|<key>[,<key>]]   SSH agent socket or keys to expose to the build (format: default\|<id>[=<socket>\|<key>[,<key>]])
  -t, --tag stringArray                               Name and optionally a tag in the 'name:tag' format
      --target string                                 Set the target build stage to build.
```

Reply:
Also, perhaps we need to check for other characters (e.g. `<`, `>`, and / or `[something]`, which _may_ be used in "usage output", but may not render well as markdown.

In that case, perhaps we should name the function `mdEscape` (without the `Pipe`) 🤔

Reply:
> Yeah I thought about that and if someone already escapes in the usage field it would look bad in the CLI so I don't think it's necessary to check that here:

Good point 👍

Reply:
> 
> 
> Also, perhaps we need to check for other characters (e.g. `<`, `>`, and / or `[something]`, which _may_ be used in "usage output", but may not render well as markdown.
> 
> In that case, perhaps we should name the function `mdEscape` (without the `Pipe`) 🤔

We could also sanitize unclosed ones in a follow-up but I think in this case we should use \`\` around like https://github.com/docker/cli-docs-tool/pull/8/files#diff-876515978c75a52e349179b237076540e109093213dc6f5f2ca06d8738e9b558R50 because Cobra already sanitizes \`\` in the CLI:

```text
      --ssh default|<id>[=<socket>|<key>[,<key>]]   SSH agent socket or keys to expose to the build (format: default|<id>[=<socket>|<key>[,<key>]])
```